src/en/nineanime/src/eu/kanade/tachiyomi/animeextension/en/nineanime/NineAnime.kt

```diff
@@ -68,67 +66,89 @@ class NineAnime : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
 
     override fun episodeListRequest(anime: SAnime): Request {
         val id = client.newCall(GET(baseUrl + anime.url)).execute().asJsoup().selectFirst("div[data-id]").attr("data-id")
-        val vrf = encodeVrf(id)
-        return GET("$baseUrl/ajax/episode/list/$id?vrf=$vrf")
+        val jsVrfInterceptor = client.newBuilder().addInterceptor(JsVrfInterceptor(id, baseUrl)).build()
+        val vrf = jsVrfInterceptor.newCall(GET("$baseUrl/filter")).execute().request.header("url").toString()
+        return GET("$baseUrl/ajax/episode/list/$id?vrf=$vrf", headers = Headers.headersOf("url", anime.url))
     }
 
     override fun episodeListParse(response: Response): List<SEpisode> {
+        val animeUrl = response.request.header("url").toString()
         val responseObject = json.decodeFromString<JsonObject>(response.body!!.string())
         val document = Jsoup.parse(JSONUtil.unescape(responseObject["result"]!!.jsonPrimitive.content))
-        return document.select(episodeListSelector()).map(::episodeFromElement).reversed()
+        val episodeElements = document.select(episodeListSelector())
+        return episodeElements.map { episodeFromElements(it, animeUrl) }
```

Comment:
maybe try to use something like `parallelMap` like in the zoro extension in order to make it faster...
code: https://github.com/jmir1/aniyomi-extensions/blob/master/src/en/zoro/src/eu/kanade/tachiyomi/animeextension/en/zoro/Zoro.kt#L408-L411

Reply:
this will make run the webview interceptor for all episodes at the same time, making it faster (in theory).
just try using parallelMap instead of map and see if it works...